fix(DH): Make font-sizes and colors same across the expandable panel

Angi-Kinas opened the pull request:
Changed the font-size and color as proposed to:
![image](https://github.com/geonetwork/geonetwork-ui/assets/133115263/83fab36a-7c6d-4fcd-8362-4ccf7105432a)



Also made opening of the panel faster.

fgravin commented:
Thanks @Angi-Kinas 
Some remarks regarding the mockup:

**1. D’où viennent ces données ?**

- Margin between labels and values are too big
- The text, before the grid is not black
![image](https://github.com/geonetwork/geonetwork-ui/assets/1491924/4767b35e-bb85-486e-8e23-fea54f9690fa)

**2. Licence ou conditions d’utilisation ?**
- Text should be small
- Would be good to have the label stuff, if possible. Meaning we should be able to categorize the licence into a label.
![image](https://github.com/geonetwork/geonetwork-ui/assets/1491924/22eda4b6-9f4d-4aee-be37-aad134039e58)

**3. Détails**
The right margin after the icon seems too big



Overall, fonts are 14px & 16px, but it's hard to understand strictly the mockup as the font is completely different :smile:

tkohr commented:
Thanks @Angi-Kinas ! I'm wondering if the font sizes are not ok in this PR though @fgravin...

> 
> **1. D’où viennent ces données ?**
> 
>     * The text, before the grid is not black

I think this will be handled by the theming config, no?
> 
> 
> **2. Licence ou conditions d’utilisation ?**
> 
>     * Text should be small

Lucile lately proposed in a JIRA comment to put it in primary color (and then same size as primary text in other panels, I guess). It would look more consistent if there was also a label inside the panel I guess. But primary in 14px won't look consistent neither I think.

jahow reviewed (approved):
All good on my side, thanks!

tkohr commented on libs/ui/elements/src/lib/metadata-info/metadata-info.component.html:
```diff
@@ -32,27 +35,30 @@
 >
-  <p class="mb-5 pt-4 whitespace-pre-line break-words" gnUiLinkify>
+  <p
+    class="text-gray-900 mb-5 pt-4 whitespace-pre-line break-words"
```
I thought, we would let the config set the `main` color here (as well as in the abstract above), but maybe I'm wrong (cc @fgravin @jahow)

fgravin commented on libs/ui/elements/src/lib/metadata-info/metadata-info.component.html:
```diff
@@ -4,11 +4,12 @@
   record.metadata.about
 </p>
-<div class="mb-6 md-description sm:mb-4 sm:pr-16" *ngIf="metadata.abstract">
+<div class="text-gray-900 mb-6 md-description sm:mb-4 sm:pr-16">
```
The text should be the `main` text color, I think it means that no color class is required.
We should not use `text-gray` things I guess.

Angi-Kinas replied:
> The text should be the `main` text color, I think it means that no color class is required. We should not use `text-gray` things I guess.

the gray-900 is derived from the main color, so that should not be a problem. the main color is almost black and here we wanted a lighter color (as you also mentioned in your review).
cc @tkohr

tkohr replied:
Ok, I thought using the `main` color would allow more consistence with other components, but looking at the card on the home page we also use the derived `text-gray-900`, so it seems ok to me, as long as we try to stay consistent with the derived grays ;-)

fgravin replied:
Why do you remove `*ngIf="metadata.abstract"` ? Is it the abstract container or not ?

And yes, the abstract should have the `main` color, and not `text-gray-900` I guess.

I think the colors are:
- Abstract title: **black**
Abstract content: **main**

- D’où viennent ces données ? title: **black**
Text underneath: **main**
Property label: **grayer**
Property color: **primary**

Angi-Kinas replied:
The ngIf was not removed but just moved to the paragraph. Otherwise the content ghost will not be shown.

Regarding the colors:
main is almost black, so compared to the mockups it would be too dark. 
![image](https://github.com/geonetwork/geonetwork-ui/assets/133115263/179bfd76-71b3-4562-96ce-2dbb2d02d87b)

Maybe we shouldn't use pure `black` but `main` because [pure black is apparently bad practice](https://uxplanet.org/basicdesign-never-use-pure-black-in-typography-36138a3327a6) and it's not dependent on the theme.

tkohr commented on libs/ui/elements/src/lib/metadata-info/metadata-info.component.html:
```diff
@@ -32,27 +35,30 @@
     <div *ngIf="metadata.recordUpdated">
       <p class="text-sm" translate>record.metadata.updatedOn</p>
-      <p class="text-primary font-medium mt-2">
+      <p class="text-primary font-medium mt-[4px]">
```
`mt-1` should be the same I think

Angi-Kinas replied:
I think it is more explixit to put it in px, because it reflects what is in the mockup. In terms of generated css it's almost the same.

tkohr replied:
Ok, so far we've privileged the non interpolated values (also for the lightest CSS possible, I think). I don't have a strong opinion on this, but maybe we should agree on a best practice.

Angi-Kinas commented:
@fgravin and @tkohr Thanks for the review. 
@fgravin I added most of your suggestions (see screenshot).

To point 2:
- As Lucile mentioned in the ticket that she wanted to change the way this is being displayed so it will be aligned with the other panels (in blue).
- the badge is only set if the flag is true. It would be too hacky to retrieve it from the text as it is right now.

Also:
Olivia and I discovered that there was another bug. The content ghost for the abstract was not shown. We fixed this as well.
